<launch>
  <!-- Arguments -->
  <arg name="model" default="$(env TURTLEBOT3_MODEL)" doc="model type [burger, waffle, waffle_pi]"/>
  <!-- <arg name="map_file" default="$(find turtlebot3_navigation)/maps2/map2.yaml"/> -->
   <arg name="map_file" default="$(find turtlebot3_navigation)/maps/map.yaml"/>
  <arg name="open_rviz" default="true"/>
  <arg name="move_forward_only" default="false"/>

  <!-- Turtlebot3 -->
  <include file="$(find turtlebot3_bringup)/launch/turtlebot3_remote.launch">
    <arg name="model" value="$(arg model)" />
  </include> 

  <!-- Map server -->
  <node pkg="map_server" name="map_server" type="map_server" args="$(arg map_file)"/>

  <!-- AMCL-->
  <!-- <include file="$(find turtlebot3_navigation)/launch/amcl.launch"/>  -->
  <node pkg="tf" type="static_transform_publisher" name="map_odom_broadcaster"  args="0 0 0 0 0 0 map odom 100"/> 


  <!-- move_base -->
  <include file="$(find turtlebot3_navigation)/launch/move_base.launch">
    <arg name="model" value="$(arg model)" />
    <arg name="move_forward_only" value="$(arg move_forward_only)"/>
  </include>

  <!-- odometry_test -->
  <!-- <node pkg="testbot" name = "odometry_test" type = "odometry_test"  output="screen" >
  </node>
  <node pkg="testbot" name = "rxST1" type = "rxST1"  output="screen" >
  </node> -->

  <!-- rviz -->
  <group if="$(arg open_rviz)"> 
    <node pkg="rviz" type="rviz" name="rviz" required="true"
          args="-d $(find turtlebot3_navigation)/rviz/turtlebot3_navigation.rviz"/>
  </group>

  <!-- <node pkg="tf" type="static_transform_publisher" name="odom_to_basefootprint"  args="0.0 0.0 0.0 0 0 0.0 /odom /base_footprint 100" /> -->

</launch>
